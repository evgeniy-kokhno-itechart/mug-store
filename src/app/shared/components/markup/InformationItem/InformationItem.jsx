import React from 'react';
import { PropTypes } from 'prop-types';
import './InformationItem.scss';

const InformationItem = ({ label, info }) => (
  <div className="information-item">
    <p className="information-item__label">
      {label}
    </p>
    <p className="information-item__info">{info}</p>
  </div>
);

InformationItem.propTypes = {
  label: PropTypes.string.isRequired,
  info: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
};

InformationItem.defaultProps = {
  info: '',
};

export default InformationItem;
