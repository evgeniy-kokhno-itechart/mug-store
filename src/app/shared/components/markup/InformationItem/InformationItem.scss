.information-item {
    display: flex;
    margin: 0.5rem 0;

    &__label {
        flex: 1;
        text-align: end;
        margin: 0 1rem;
        font-weight: 500;
    }

    &__info {
        flex: 1;
        margin: 0 1rem;
    }
}