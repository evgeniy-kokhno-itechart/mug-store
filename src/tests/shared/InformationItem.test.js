/* eslint-disable no-undef */
import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import InformationItem from '../../shared/markup-usage/InformationItem';

configure({ adapter: new Adapter() });

describe('<InformationItem />', () => {
  let InformationItemWrapper;

  beforeEach(() => {
    InformationItemWrapper = shallow(<InformationItem label="test label" />);
  });

  it('check InformationItem renders properly with default props', () => {
    expect(InformationItemWrapper.debug()).toMatchSnapshot();
  });

  it('check InformationItem renders properly when info is number', () => {
    InformationItemWrapper.setProps({ info: 1 });
    expect(InformationItemWrapper.debug()).toMatchSnapshot();
  });

  it('check InformationItem renders properly when info is string', () => {
    InformationItemWrapper.setProps({ info: 'test info' });
    expect(InformationItemWrapper.debug()).toMatchSnapshot();
  });
});
