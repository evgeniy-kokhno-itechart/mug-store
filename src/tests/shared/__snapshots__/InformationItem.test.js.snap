// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InformationItem /> check InformationItem renders properly when info is number 1`] = `
"<div className=\\"row justify-content-center mx-3 my-2 mx-md-2\\">
  <span className=\\"col-md-4 text-md-right px-0 px-md-1\\">
    <b>
      test label
    </b>
  </span>
  <span className=\\"col-md-4 text-md-left px-0 px-md-1\\">
    1
  </span>
</div>"
`;

exports[`<InformationItem /> check InformationItem renders properly when info is string 1`] = `
"<div className=\\"row justify-content-center mx-3 my-2 mx-md-2\\">
  <span className=\\"col-md-4 text-md-right px-0 px-md-1\\">
    <b>
      test label
    </b>
  </span>
  <span className=\\"col-md-4 text-md-left px-0 px-md-1\\">
    test info
  </span>
</div>"
`;

exports[`<InformationItem /> check InformationItem renders properly with default props 1`] = `
"<div className=\\"row justify-content-center mx-3 my-2 mx-md-2\\">
  <span className=\\"col-md-4 text-md-right px-0 px-md-1\\">
    <b>
      test label
    </b>
  </span>
  <span className=\\"col-md-4 text-md-left px-0 px-md-1\\" />
</div>"
`;
